### PR TITLE
Log Update: Rounding off the PCO2 values to nearest integer

### LINKED ISSUE #8593: ABG Log update -PCO2 decimal number entry not allowed.
On doing detailed log update of ABG Test.
All the other fields except Pco2  allow decimal values,Pco2 doesn't allow decimal values entry.
**To Reproduce**

1. Go to a patient dashboard 
2. Click on Log update 
3. Round type Detailed update
4. Update ARTERIAL BLOOD GASES Report.
5. All the field except Pco2 allowed decimal value entry.
6. Unable to save the data of not integer value.

### PR DESCRIPTION
**Pull request by Jacobjeevan**
## Proposed Changes

Backend doesn't accept decimal values for certain fields, so rounding the values to nearest integer instead.

- Fixes #8593 

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Form/FormFields/RangeFormField.tsx [hunk 1]
```diff
@@ -18,6 +18,7 @@ type BaseProps = FormFieldBaseProps<number> & {
   valueDescriptions?: ValueDescription[];
   hideInput?: boolean;
   hideUnitInLabel?: boolean;
+  allowIntegersOnly?: boolean;
```
Comment (rithviknishad) [inline src/Components/Form/FormFields/RangeFormField.tsx]: just a thought, we could automatically infer whether to allow integers only or not by checking against `step` value right?

**Comment by nihal467** [comment]
@rithviknishad @Jacobjeevan once the PR is read for review, post a comment or change the label to need testing

File: src/Components/Form/FormFields/RangeFormField.tsx (hunk 1 above)
Comment (Jacobjeevan) [reply src/Components/Form/FormFields/RangeFormField.tsx]: 🤔  yea that will work for almost all fields (except insulin intake dosage and systolic/diastolic, so we can either just add step of 0.1 there, or disallow decimal entries for the field).

Comment (Jacobjeevan) [reply src/Components/Form/FormFields/RangeFormField.tsx]: @rithviknishad what do you suggest? ^

Comment (rithviknishad) [reply src/Components/Form/FormFields/RangeFormField.tsx]: we can set the step `0.1` for that field.

**Comment by bodhish** [comment]
@Jacobjeevan you need to add `need testing` label for it to be tested.

**Comment by samholics** [comment]
regarding this issue, Actually the PCo2 and Po2 fields needs at least two decimal places. @nihal467 should i file an another issue or can we do it here itself:? the same fields needs another option to switch  between "mmHg" and "Kpa" units. currently there is only "mmHg" unit.  both units needs atleast two decimal places. also "pH" needs three decimal places. refer the below abg results
eg: 
image
image

**Comment by samholics** [comment]
> regarding this issue, Actually the PCo2 and Po2 fields needs at least two decimal places. @nihal467 should i file an another issue or can we do it here itself:? the same fields needs another option to switch between "mmHg" and "Kpa" units. currently there is only "mmHg" unit. both units needs atleast two decimal places. also "pH" needs three decimal places. refer the below abg results eg: image image

@nihal467 talked with aparna and filing this as a new issue

**Comment by nihal467** [comment]
@Jacobjeevan 

- [ ] we should allow PCO2 to have decimal values per the issue. Following the standard, allow up to two decimal points for the PCO2. Please make the changes in this PR and update the backend accordingly to support it.

**Comment by Jacobjeevan** [comment]
> @Jacobjeevan
> 
>     * [ ]  we should allow PCO2 to have decimal values per the issue. Following the standard, allow up to two decimal points for the PCO2. Please make the changes in this PR and update the backend accordingly to support it.

Hmm, Sam already filed an issue for this and looks like someone is working on it 🤔 #8858  What do you think? Let them handle it, or should I just make the modification in this PR itself? @nihal467

**Comment by bodhish** [comment]
@Jacobjeevan lets get this merged

**Comment by Jacobjeevan** [comment]
> @Jacobjeevan lets get this merged

It's good to go, since @abhip161 is working on #8858 and #8859 , I'll let them handle the rest 👍

**Comment by nihal467** [comment]
LGTM

**Review by bodhish — APPROVED** [review]

**Pull request merged**